### PR TITLE
[Documentation] Customization Guide: State Machine

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | no
| New docs?    | yes
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

### CODE REVIEW

File: docs/customization/state_machine.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                    sylius_update_exchange_rate:
+                        on: ["select_payment"]
+                        # here you can change what the callback will do
+                        do: ["@service", "update"]
+                        args: ["object"]
+
+How to disable a callback?
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you would like to turn off a callback of a state machine you nee to set its ``disabled`` option to true.
```
Comment (tuka217) [inline docs/customization/state_machine.rst]: (...)you **need** to(...)

File: docs/customization/state_machine.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+.. code-block:: yaml
+
+    # app/config/state_machine.yml
+    winzou_state_machine:
+        sylius_order_checkout:
+            transitions:
+                cancel:
+                    from: [completed]
+                    to: payment_selected
```
Comment (pjedrzejewski) [inline docs/customization/state_machine.rst]: These examples need to make some sense. :D

File: docs/customization/state_machine.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+Let's assume that you would like to add a new **callback** to the state machine of :doc:`Sylius' Checkout </book/checkout>`,
+that will do something on an already defined transition.
+
+You will need to add these few lines to the ``state_machine.yml``:
+
+.. code-block:: yaml
+
+    # app/config/state_machine.yml
+    winzou_state_machine:
+        sylius_order_checkout:
```
Comment (pjedrzejewski) [inline docs/customization/state_machine.rst]: This example is perfect, but it should be ``sylius_order`` state machine, not checkout.

File: docs/customization/state_machine.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+On the example of the state machine of :doc:`Sylius' Checkout </book/checkout>`, we can turn off the ``sylius_update_exchange_rate`` callback:
+
+.. code-block:: yaml
+
+    # app/config/state_machine.yml
+    winzou_state_machine:
+        sylius_order_checkout:
+            callbacks:
+                after:
+                    sylius_update_exchange_rate:
```
Comment (pjedrzejewski) [inline docs/customization/state_machine.rst]: I'd use email callback as example, these are simplest and real world use cases.

**Comment by pjedrzejewski** [comment]
Thank you Magda! :)

**Pull request merged**